compiler/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -170,15 +169,14 @@ impl<'interner> TypeChecker<'interner> {
                 });
 
                 // Check that we are not passing a mutable reference from a constrained runtime to an unconstrained runtime
-                for (typ, _, _) in args.iter() {
-                    if is_current_func_constrained
-                        && is_unconstrained_call
-                        && matches!(&typ, Type::MutableReference(_))
-                    {
-                        self.errors.push(TypeCheckError::ConstrainedReferenceToUnconstrained {
-                            span: self.interner.expr_span(expr_id),
-                        });
-                        return Type::Error;
+                if is_current_func_constrained && is_unconstrained_call {
+                    for (typ, _, _) in args.iter() {
+                        if matches!(&typ, Type::MutableReference(_)) {
```

Comment:
```suggestion
                        if matches!(&typ.follow_bindings(), Type::MutableReference(_)) {
```
Didn't catch this in the first PR but we should always call `follow_bindings` before checking a type is a certain variant.